contracts/GenArt721CoreV3_Engine_Flex_PROOF.sol

```diff
@@ -0,0 +1,2278 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.17;
+
+// Created By: Art Blocks Inc.
+
+import "./interfaces/0.8.x/IRandomizerV2.sol";
+import "./interfaces/0.8.x/IAdminACLV0.sol";
+import "./interfaces/0.8.x/IEngineRegistryV0.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3_Engine_Flex.sol";
+import "./interfaces/0.8.x/IDependencyRegistryCompatibleV0.sol";
+import "./interfaces/0.8.x/IManifold.sol";
+
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+import "./libs/0.8.x/ERC721_PackedHashSeed.sol";
+import "./libs/0.8.x/BytecodeStorage.sol";
+import "./libs/0.8.x/Bytes32Strings.sol";
+
+/**
+ * @title Art Blocks Engine Flex ERC-721 core contract, V3.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * This contract is designed to be managed, with progressively limited powers
+ * as a project progresses from active to locked.
+ * Privileged roles and abilities are controlled by the admin ACL contract and
+ * artists. Both of these roles hold extensive power and can arbitrarily
+ * control and modify portions of projects, dependent upon project state. After
+ * a project is locked, important project metadata fields are locked including
+ * the project name, artist name, and script and display details. Edition size
+ * can never be increased.
+ * Care must be taken to ensure that the admin ACL contract and artist
+ * addresses are secure behind a multi-sig or other access control mechanism.
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to the Admin ACL contract:
+ * - updateArtblocksDependencyRegistryAddress
+ * - updateProviderSalesAddresses
+ * - updateProviderPrimarySalesPercentages (up to 100%)
+ * - updateProviderSecondarySalesBPS (up to 100%)
+ * - updateMinterContract
+ * - updateRandomizerAddress
+ * - toggleProjectIsActive
+ * - addProject
+ * - forbidNewProjects (forever forbidding new projects)
+ * - updateDefaultBaseURI (used to initialize new project base URIs)
+ * - updateIPFSGateway
+ * - updateArweaveGateway
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to either the the Artist address or
+ * the Admin ACL contract, only when the project is not locked:
+ * - updateProjectName
+ * - updateProjectArtistName
+ * - updateProjectLicense
+ * - Change project script via addProjectScript, updateProjectScript,
+ *   and removeProjectLastScript
+ * - updateProjectScriptType
+ * - updateProjectAspectRatio
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to only the Artist address:
```

Comment:
Let's update this section's description to be something like `The following functions are restricted to only the Artist or Admin ACL contract of a valid project ID:`

Additionally, since the `onlyArtist` implicitly checked for a valid project ID, please add `_onlyValidProjectId(_projectId)` to each of the adjusted functions in this section 🙏

Reply:
Added the former. Also happy to add the latter, however `onlyArtist` does not imply `_onlyValidProjectId` as far as I can tell from https://github.com/ArtBlocks/artblocks-contracts/blob/b2c4e24848fee440226bc56302edc1c72c2c5b3f/contracts/GenArt721CoreV3_Engine_Flex.sol#L329-L345

Reply:
resolved in 4b7f5bb & 48be657 🎉